packages/editor/src/store/actions.js

```diff
@@ -362,9 +380,12 @@ export function* savePost( options = {} ) {
 			method,
 			data: toSend,
 		} );
-		const resetAction = isAutosave ? 'resetAutosave' : 'resetPost';
 
-		yield dispatch( STORE_KEY, resetAction, newPost );
+		if ( isAutosave ) {
+			yield dispatch( 'core', 'receiveAutosaves', post.id, newPost );
```

Comment:
`receiveAutosaves` expects an array of autosaves, but we're passing the singular `newPost`. Will that work?

**Edit:**  Yeah, I think this is not working correctly, as it changes the structure of `autosaves[ postId ]` from an array to an object:

![image](https://user-images.githubusercontent.com/1779930/56161897-b4882600-5f98-11e9-80ec-ab14bd209b04.png)

Reply:
Great spot. I've made `receiveAutosaves` so that it can accept either an array or a singular object. Added a test case for the conversion from single autosave to an array.